target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22.dts

```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "vr9.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mips/lantiq_rcu_gphy.h>
+
+/ {
+	compatible = "arcadyan,vgv953akw22", "lantiq,xway", "lantiq,vr9";
+	model = "Speedport W 921V";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	aliases {
+		led-boot = &led_power_white;
+		led-failsafe = &led_power_red;
+		led-running = &led_power_white;
+		led-upgrade = &led_waiting;
+
+		led-dsl = &led_dsl;
+		led-internet = &led_internet;
+		led-wifi = &led_wifi;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		wifi {
+			label = "wifi";
+			gpios = <&gpio 29 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+
+		dect_paging {
+			label = "dect_paging";
+			gpios = <&gpio 40 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_PHONE>;
+		};
+
+		dect_pairing {
+			label = "dect";
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_CONFIG>;
+		};
+
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_waiting: waiting {
+			label = "vgv953akw22:yellow:waiting";
+			gpios = <&stp 16 GPIO_ACTIVE_LOW>;
+		};
+		led_service: service {
+			label = "vgv953akw22:yellow:sevice";
+			gpios = <&stp 17 GPIO_ACTIVE_LOW>;
+		};
+		led_phone: phone {
+			label = "vgv953akw22:white:phone";
+			gpios = <&stp 18 GPIO_ACTIVE_LOW>;
+		};
+		led_wifi: wifi {
+			label = "vgv953akw22:white:wifi";
+			gpios = <&stp 19 GPIO_ACTIVE_LOW>;
+		};
+		led_internet: internet {
+			label = "vgv953akw22:white:internet";
+			gpios = <&stp 20 GPIO_ACTIVE_LOW>;
+		};
+		led_dsl: dsl {
+			label = "vgv953akw22:white:dsl";
+			gpios = <&stp 21 GPIO_ACTIVE_LOW>;
+		};
+		led_power_red: power_red {
+			label = "vgv953akw22:red:power";
+			gpios = <&stp 22 GPIO_ACTIVE_LOW>;
+		};
+		led_power_white: power_white {
+			label = "vgv953akw22:white:power";
+			gpios = <&stp 23 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+	};
+
+	usb0_vbus: regulator-usb0-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB0_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
```

Comment:
See USB comment further down only needed for "gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;"

Reply:
Why do you need the GPIO to be high?
I'm just commenting here, because I had a similar problem on the AVM FRITZ!3390. I added a `gpio-hog` to force the GPIO to a predefined state, but on the Lantiq you need one more modification to `vr9.dtsi` which is also included in #2662, but not yet merged.

Reply:
GPIO 32 and 33 are necessary to power the USB Ports, only usb_phy1 with 2 Ports.
If one of the GPIO's is high the port is not supplied with 5V.
it looks nicer with the gpio-hog, found your commit earlier but overlooked it. Thanks for mentioning it, this requires either your commit to be merged or added to this dts file.
ifxhcd-rst-dev or ifxhcd-pwr would be an appropriate name?